src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,134 @@
+import { useState } from "react";
+import { Button, Card, Col, Form, Row } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import { MovieCard } from "../movie-card/movie-card";
+
+export const ProfileView = ({ movies, user }) => {
+  const storedToken = localStorage.getItem("token");
+
+  if (!user) {
+    console.log("User not found in localStorage");
+    return;
+  }
+
+  if (!storedToken) {
+    console.log("Token not found in localStorage");
+    return;
+  }
+
+  const [name, setName] = useState(user?.Name || "");
+  const [email, setEmail] = useState(user?.Email || "");
+  const [birthday, setBirthday] = useState(user?.Birthday || "");
+  const [password, setPassword] = useState(user?.Password || "");
+
+  const fav = movies.filter((movie) => {
+    return user.FavoriteMovies.includes(movie._id);
+  });
+
+  const handleToggleFavorite = (updatedUserDetails) => {
+    localStorage.setItem("user", JSON.stringify(updatedUserDetails));
+  };
```

Comment:
removing a movie from favorites requires upload in order to see changes. You can fix it by changing this function and take as a prop from main-view component(just like what you did with the movie-card component).